libs/shared/src/lib/modules/markdown/markdown.component.ts

```diff
@@ -0,0 +1,59 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ViewChild,
+    forwardRef
+} from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { MarkdownButtonSyntax } from './components';
+import { MarkdownInputDirective } from './directives';
+import { MarkdownInputState, MarkdownViewModeState } from './states';
+
+@Component({
+    selector: 'acua-markdown [formControl]',
+    templateUrl: './markdown.component.html',
+    styleUrls: ['./markdown.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => MarkdownComponent),
+            multi: true
+        }
+    ]
+})
+export class MarkdownComponent implements ControlValueAccessor {
```

Comment:
```suggestion
export class MarkdownEditorComponent implements ControlValueAccessor {
```

А модуль - MarkdownModule, який об'єднує Editor + Preview головні компоненти, для використання ззовні